### PR TITLE
Use Mix_LoadMUS_RW() instead of saving MIDI to disk

### PR DESCRIPTION
**Pull request by erysdren**
Whenever `MUSIC_PlaySong()` was written, `Mix_LoadMUS_RW()` didn't exist. But now it does! So let's use it.

### CODE REVIEW

File: rott/dukemusc.c [hunk 1]
```diff
@@ -265,65 +266,33 @@ int MUSIC_StopSong(void)
...
 
-    // save the file somewhere, so SDL_mixer can load it
-    filename = M_StringJoin(ApogeePath, PATH_SEP_STR, "tmpsong.mid", NULL);
-    handle = SafeOpenWrite(filename);
-    
-    SafeWrite(handle, song, size);
-    close(handle);
-    
-    music_songdata = song;
+	// create rw
```
Comment (fabiangreffrath) [inline rott/dukemusc.c:277]: Could you add a check for `size == 0` and return early?

File: rott/dukemusc.c [hunk 2]
```diff
@@ -265,65 +266,33 @@ int MUSIC_StopSong(void)
...
+	SDL_RWops *rw = SDL_RWFromConstMem(song, size);
+	if (rw == NULL) {
+		return MUSIC_Error;
+	}
 
-    // finally, we can load it with SDL_mixer
-    music_musicchunk = Mix_LoadMUS(filename);
-    free(filename);
+    // load with SDL_mixer
+    music_musicchunk = Mix_LoadMUS_RW(rw, SDL_TRUE);
```
Comment (fabiangreffrath) [inline rott/dukemusc.c:284]: The `SDL_TRUE` in the second parameter means that we don't have to care about freeing the `rw` stuff ourselves, right?

Comment (erysdren) [reply rott/dukemusc.c:284]: Correct

File: rott/dukemusc.c (hunk 1 above)
Comment (erysdren) [reply rott/dukemusc.c:277]: Will do.

**Review by fabiangreffrath — APPROVED** [review]
Thank you!

**Comment by fabiangreffrath** [comment]
Yep, right.

**Pull request merged**